src/log.py

```diff
@@ -0,0 +1,94 @@
+"""
+log
+
+By: Norma Chang
+"""
+import json
+import time
+import os.path
+
+
+class Log:
+    """
+    Logging class.
+
+    Logs data from the properties of an instance of an vehicle class for
+    testing purposes.
+    """
+    def __init__(self):
+
+        self.file_name = ""
+        self.class_name = ""
+        self._start_time = time.strftime('%b %d %Y %H:%M:%S', time.gmtime())
+        self._start_time = self._start_time.replace(" ", "").replace(":", "")
+
+
+    def isFollower(self, vehicle):
+        if vehicle.__class__.__name__ == 'Follower':
+            return True
+        else:
+            return False
+
+
+    def isLeader(self, vehicle):
+        if vehicle.__class__.__name__ == 'Leader':
+            return True
+        else:
+            return False
+
+
+    def log_vehicle_data(self, vehicle):
+        """
+        Logs vehicle data.
+
+        :param vehicle: The instance of the vehicle class
+        :type vehicle: object
+        """
+
+        if self.isFollower(vehicle):
+            self.class_name = vehicle.__class__.__name__
+            self.vehicle_dict = {
+                'Time': time.time(),
+                'Distance': vehicle._distance,
+                'Speed': vehicle._speed,
+                'Turn Angle': vehicle._turn_angle,
+                'Yaw': vehicle._yaw,
+            }
+        elif self.isLeader(vehicle):
+            self.class_name = vehicle.__class__.__name__
+            self.vehicle_dict = {
+                'Time': time.time(),
+                'Input': vehicle.turn(position),
+                'Speed': vehicle.set_speed(position),
+            }
+        else:
+            raise ValueError('Unexpected class type')
+
+        self.file_name = self.class_name + self._start_time + '.log'
+        self.write_to_file()
+
+
+    def write_to_file(self):
+        """
+        Write vehicle data into text file.
+        """
+        with open(self.file_name, 'a') as outfile:
+            json.dump(self.vehicle_dict, outfile)
+            outfile.write(',')
+
+
+    def read_log_file(self, file_name):
+        """
+        Reads the log file.
+
+        :param: file_name: File name is generated in log_vehicle_data
+                           after verifying the type of vehicle.
```

Comment:
typo, but also too much info. Why do we specify that it's a file generated in log_vehicle_data? Shouldn't it just work with any file containing dictionaries?

Reply:
I'm just specifiying where the name of the file coming from, I can make it more general.